q2_longitudinal/_utilities.py

```diff
@@ -392,29 +402,52 @@ def _control_chart_subplots(state_column, metric, metadata, group_column,
         c, gm, gs = _control_chart(
             state_column, metric, group_md, None, ci=ci, legend=False,
             color=next(colors), plot_control_limits=True, ax=axes[num + 1],
-            palette=None)
+            palette=None, xtick_interval=xtick_interval)
         c.set_title('{0}: {1}'.format(group_column, group))
+        c = _set_xticks(c, group_md, state_column, states, xtick_interval)
     plt.tight_layout()
 
     return chart, global_mean, global_std
 
 
+def _set_xtick_interval(xtick_interval, states):
+    if xtick_interval is None:
+        if len(states) > 30:
+            xtick_interval = ceil(len(states) / 30)
+        else:
+            xtick_interval = 1
+    return xtick_interval
+
+
+def _set_xtick_labels(metadata, state_column, states, xtick_interval):
+    # pull x-axis labels from array of unique states, slice xtick_interval
+    x_tick_labels = {state: metadata[metadata[state_column] == state]
+                     for state in states[::xtick_interval]}
+    # sort labels by key (state), add sample size to labels.
+    # x_tick_labels[0] must be empty because this label does not appear in plot
```

Comment:
To be honest I am not 100% sure... as far as I can tell, ln 438:
```
ax.xaxis.set_major_locator(ticker.MultipleLocator(xtick_interval))
```
causes the view limits to bump. Feeding text labels to a continuous x-axis causes the labels to decouple from the data, which is bad but inevitable (since we are adding sample sizes here). Just setting the ticks directly (without ``MultipleLocator``) would be ideal, but is needed here to locate the correct label position if we set `xtick_interval` > 1.

(an alternative for ln 438 is to set the xticks with something like:
```
    ax.set_xticks(np.arange(
        int(sorted(states)[0]), int(sorted(states)[-1]) + 1, xtick_interval))
```
but then we need to reset the xlimits and it just gets messy)